### ISSUE TITLE
Initial window scaling factor functionality documentation

### ISSUE THREAD

**Issue by wcout** [fltk/fltk#152]
Up to now I was convinced that setting the FLTK_SCALING_FACTOR environment variable overrides all system specific scaling detection. But now I see (on X11) that this value is multiplied into any system specific value to set the scale for a window.
Is this the way it is meant to work?

The documentation states:

"The scaling factor is initialized by the library to a value based on information obtained from the OS. If this initial value is not satisfactory, the FLTK_SCALING_FACTOR environment variable can be set to a value FLTK will use as initial scaling factor."

Currently it would be more precise to say, that the value of FLTK_SCALING_FACTOR is applied as an additional factor to any factor based on information from the OS.

**Comment by ManoloFLTK** [comment]
Yes, it was decided that the FLTK_SCALING_FACTOR environment variable would modulate the OS-chosen
initial value.

However, that other API:  void Fl::screen_scale(int n, float f)
    https://fltk.gitlab.io/fltk/group__fl__screen.html#ga0ce8513415bdba5bd2d2f7661b5bba5d
can be used to set the scaling factor of screen #n unconditionally to the value f.

**Comment by wcout** [comment]
Seems I missed that change and have since not been on a system with different dpi settings than 96. 
Thanks, I'll use this other functionality.
Do you agree the current description is not clear enough in this aspect?

**Comment by ManoloFLTK** [comment]
I have just committed a doc change to improve that.
Can you review it, please, and close this issue accordingly.
TIA

P.S. I'm very interested in any feedback about using FLTK 1.4 on a High-DPI Linux system.

**Comment by wcout** [comment]
The header section of "Screen functions" now is clear.
But I quite don't understand the description of screen_scale(int n, float factor):

> Set the value of the GUI scaling factor for screen number n.
>
>When this function is called before the first window is show()'n it sets the application's initial scaling factor value. Otherwise, it sets the scale factor value of all windows mapped to screen number n

The "otherwise" puzzles mes. Does that mean if called after the first window is show, it affects only the currently open windows and keeps the initial default?
Or does it mean it changes the initial and future factor and applies the scale to all mapped windows too?

Also is this really before the first window is shown or does it suffice to open the display?

In short: At application start, which is the point where calling scale_factor() changes the factor for all subsequent windows on this screen?

**Comment by ManoloFLTK** [comment]
The description of the Fl::screen_scale(int n, float factor) function has been
simplified. It's now :
   Sets the value of the GUI scaling factor for screen number n.
   Also sets the scale factor value of all windows mapped to screen number n, if any.

Thus, this function unconditionally sets the scaling factor value of screen #n.
And, if there are FLTK windows mapped on that screen, it also sets their scaling
factor to that same value. If there is now mapped window, the effect of the
Fl::screen_scale() call manifests when a window gets mapped on it.
